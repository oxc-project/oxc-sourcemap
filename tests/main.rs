use std::fs;

use oxc_sourcemap::{SourceMap, SourcemapVisualizer, Token, Tokens};

#[test]
fn snapshot_sourcemap_visualizer() {
    insta::glob!("fixtures/**/*.js", |path| {
        let js = fs::read_to_string(path).unwrap();
        let js_map = fs::read_to_string(path.with_extension("js.map")).unwrap();
        let sourcemap = SourceMap::from_json_string(&js_map).unwrap();
        let visualizer = SourcemapVisualizer::new(&js, &sourcemap);
        let visualizer_text = visualizer.get_text();
        insta::with_settings!({ snapshot_path => path.parent().unwrap(), prepend_module_to_snapshot => false, snapshot_suffix => "", omit_expression => true }, {
            insta::assert_snapshot!("visualizer", visualizer_text);
        });
    });
}

#[test]
fn invalid_token_position() {
    let mut tokens = Tokens::new();
    tokens.push(Token::new(0, 0, 0, 0, Some(0), None));
    tokens.push(Token::new(0, 10, 0, 0, Some(0), None));
    tokens.push(Token::new(0, 0, 0, 10, Some(0), None));

    let sourcemap = SourceMap::new(
        None,
        vec![],
        None,
        vec!["src.js".into()],
        vec![Some("abc\ndef".into())],
        tokens,
        None,
    );
    let js = "abc\ndef\n";
    let visualizer = SourcemapVisualizer::new(js, &sourcemap);
    let visualizer_text = visualizer.get_text();
    insta::assert_snapshot!("invalid_token_position", visualizer_text);
}
