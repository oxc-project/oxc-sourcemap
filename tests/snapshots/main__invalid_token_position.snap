---
source: tests/main.rs
expression: visualizer_text
snapshot_kind: text
---
- src.js
(0:0) "abc\n" --> (0:0) "abc\n"
(0:0) --> (0:10) [invalid]
(0:10) [invalid] --> (0:0)
